src/api/api.c

```diff
@@ -100,16 +100,68 @@ int8_t z_view_keyexpr_from_str_unchecked(z_view_keyexpr_t *keyexpr, const char *
     return _Z_RES_OK;
 }
 
-int8_t z_keyexpr_to_string(const z_loaned_keyexpr_t *keyexpr, z_owned_string_t *s) {
+int8_t z_keyexpr_as_view_string(const z_loaned_keyexpr_t *keyexpr, z_view_string_t *s) {
     int8_t ret = _Z_RES_OK;
-    s->_val = _z_string_make(keyexpr->_suffix);
-    if (!_z_string_check(&s->_val)) {
-        ret = _Z_ERR_SYSTEM_OUT_OF_MEMORY;
+    if (keyexpr->_id == Z_RESOURCE_ID_NONE) {
+        s->_val = _z_string_wrap(keyexpr->_suffix);
+    } else {
+        ret = _Z_ERR_GENERIC;
     }
 
     return ret;
 }
 
+int8_t z_keyexpr_concat(z_owned_keyexpr_t *key, const z_loaned_keyexpr_t *left, const char *right, size_t len) {
+    z_keyexpr_null(key);
+    if (len == 0) {
+        return z_keyexpr_clone(key, left);
+    } else if (right == NULL) {
+        return _Z_ERR_INVALID;
+    }
+    size_t left_len = strlen(left->_suffix);
+    if (left_len == 0) return _Z_ERR_INVALID;
```

Comment:
IMHO we should use this style (and AFAIK it is required for MISRA C etc.)
```suggestion
    if (left_len == 0) {
        return _Z_ERR_INVALID;
    }
```